File: api/src/main/java/com/gkh/syntheticmonitor/model/SMExecutionContext.java
package com.gkh.syntheticmonitor.model;

import com.jayway.jsonpath.JsonPath;
import lombok.Data;

import java.util.HashMap;

@Data
public class SMExecutionContext {
	private Report report = new Report();
	private HashMap vars = new HashMap();
	private String status;
	private String content;
	private String contentType;

	/**
	 *
	 * @param path
	 * @return
	 */
	public String jsonPath(String path) {
		var jsonContext =  JsonPath.parse(this.content);
		String value = jsonContext.read(path);
		return value;
	}

}

File: api/src/main/kotlin/com/gkh/syntheticmonitor/model/SMExecutionContext.kt
package com.gkh.syntheticmonitor.model

import com.jayway.jsonpath.JsonPath
import lombok.Data
import java.util.*

class SMExecutionContext {

    val report = Report()
    var vars: HashMap<String,String> = HashMap()
    var status: String? = null
    var content: String? = null
    var contentType: String? = null

    /**
     *
     * @param path
     * @return
     */
    fun jsonPath(path: String?): String {
        val jsonContext = JsonPath.parse(content)
        return jsonContext.read(path)
    }
}